segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/ExtendedS3IntegrationTest.java

```diff
@@ -0,0 +1,130 @@
+        URI uri = URI.create(endpoint);
+        baseDir = Files.createTempDirectory("extendeds3_wrapper").toString();
+        this.configBuilder.include(ExtendedS3StorageConfig.builder()
+                                                          .with(ExtendedS3StorageConfig.BUCKET, "kanpravegatest")
+                                                          .with(ExtendedS3StorageConfig.ACCESS_KEY_ID, "x")
+                                                          .with(ExtendedS3StorageConfig.SECRET_KEY, "x")
+                                                          .with(ExtendedS3StorageConfig.URI, endpoint));
+    }
+
+    /**
+     * Shuts down BookKeeper and cleans up file system directory
+     */
+    @After
+    public void tearDown() throws Exception {
+        bookkeeper.close();
+        FileHelpers.deleteFileOrDirectory(new File(baseDir));
+    }
+
+    //endregion
+
+    //region StreamSegmentStoreTestBase Implementation
+
+    @Override
+    protected ServiceBuilder createBuilder(ServiceBuilderConfig builderConfig) {
+        return ServiceBuilder
+                .newInMemoryBuilder(builderConfig)
+                .withCacheFactory(setup -> new RocksDBCacheFactory(builderConfig.getConfig(RocksDBConfig::builder)))
+                .withStorageFactory(setup -> {
+                    StorageFactory f = new LocalExtendedS3StorageFactory(
+                            setup.getConfig(ExtendedS3StorageConfig::builder),
+                            this.aclMap,
+                            setup.getExecutor());
+                    return new ListenableStorageFactory(f);
+                })
+                .withDataLogFactory(setup -> new BookKeeperLogFactory(setup.getConfig(BookKeeperConfig::builder),
+                        bookkeeper.getZkClient(), setup.getExecutor()));
+    }
+
+
+    /*
+     * We are declaring a local factory here because we need a factory that creates adapters that interact
+     * with the local file system for the purposes of testing. Ideally, however, we should mock the extended
+     * S3 service rather than implement the storage functionality directly in the adapter.
+     */
+    private class LocalExtendedS3StorageFactory implements StorageFactory {
+
+        private final ExtendedS3StorageConfig config;
+        private final AclMap aclMap;
+        private final ExecutorService executor;
+
+        public LocalExtendedS3StorageFactory(ExtendedS3StorageConfig config,
+                                             AclMap aclMap,
+                                             ExecutorService executor) {
+            Preconditions.checkNotNull(config, "config");
+            Preconditions.checkNotNull(executor, "executor");
+            this.config = config;
+            this.aclMap = aclMap;
+            this.executor = executor;
+        }
+
+        @Override
+        public Storage createStorageAdapter() {
+            URI uri = URI.create("http://localhost:9020");
```

Comment:
you are hardcoding the port here. You need to store your randomly assigned port into a variable (where you do it above) and then reuse it here.